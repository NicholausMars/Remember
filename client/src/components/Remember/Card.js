import React from  'react';
import PropTypes from 'prop-types';
/*
  @params key object
*/
// props.card is sent so i just take off a bit of it
export const Card = ({title, description}) => {
  return (
    <div className="container-card">
      <div className="card">
        <div className="card__side  card__side--front">
          <div className="card__title card__title--center">{title}</div>
        </div>
        <div className="card__side card__side--back">
          <div className="card__title card__title--border">{title}</div>
          <div className="card__description">
            {description}
          </div>
        </div>
      </div>
    </div>
  );
}

Card.propTypes ={
  title: PropTypes.string,
  description: PropTypes.string
}

  //{/* I want a image on the front part of this card to relate to the matterial in which they want to remember*/}
