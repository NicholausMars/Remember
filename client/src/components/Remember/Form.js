import React, {Component} from 'react';
import { connect } from 'react-redux';
import { addCard } from '../../actions';
import  uuidv1 from 'uuid/v1'
import PropTypes from 'prop-types';
/*
  I want the update method to use the same form.
   I want to make it obvious that the forms beening use for the update
*/

export class Form extends Component{
  constructor(props){
    super(props);
    this.state = {
      title:'',
      description:''
    };
  }

  Add = () => {

    if(validate(this.state))
      this.props.addCard({id:uuidv1() ,...this.state});
      this.emptyState();
  }
  emptyState = () => {
    this.setState({  title: '',description: ''});
  }

  render(){
    // const {display, id, title, description} = this.props.formupdate;
    // console.log(this.props.formupdate);
    return (
           <div>
                <div className="input__box">
                  <input
                    autoFocus
                    className="input__field"
                    type="text"
                    onChange={(event)=>{ this.setState({ title: event.target.value}) }}
                    value={this.state.title}
                  />
                  <textarea
                    className="input__textarea"
                    name="basic" rows="5" cols="50" maxLength="200"
                    onChange={(event)=>{ this.setState({ description: event.target.value }) }}
                    value={this.state.description}
                    >
                  </textarea>
                  {
                    /*
                    steps:
                    1.render button
                    2. when button is click
                    3. update forUpdate display: false card to be null
                    4. use the update card method to remove and add a new card
                    5. end of update
                    */
                  }
                  <button onClick={this.Add}  className="btn btn__add">+</button>
                </div>
          </div>
        );
  }
}
Form.propTypes ={
  addCard: PropTypes.func
}

export const mapStateToProps = (state) => {
  return {
    formupdate: state.formupdate
  };
}
export const validate = ({title,description }) => {
    if( title !== '' && description !== '') return true;
    return false;
}
export default connect(mapStateToProps, {addCard})(Form);

//https://medium.com/backticks-tildes/testing-your-react-component-with-jest-and-enzyme-276eef45bea0
