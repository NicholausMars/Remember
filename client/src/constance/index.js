export const CHECKBOX = "checkbox",
             TEXT     = "text",
             FILE     = "file",
             SELECT   = "select",
             PASSWORD = "password";
