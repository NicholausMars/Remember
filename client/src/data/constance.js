export const INPUT = 'Stay Positive';
export const TEXT_AREA = 'Can Find The Thing that drives you crazy in this world and then you will be happy';
export const Rememberprops = {cards: [
  {title:'TITLE1', description:'DESCRIPTION1'},
  {title:'TITLE2', description:'DESCRIPTION2'}
]};
