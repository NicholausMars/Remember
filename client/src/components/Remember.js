import React, {Component} from 'react';
import { connect } from 'react-redux';
import {Card} from './Card';
import Form from './Form';
import PropTypes from 'prop-types';

export class Remember extends Component{
  render(){
    return (
      <div>
        <h1 className="heading-primary">Remember</h1>
        <Form />
        <hr className="u-mt-sm u-mb-sm"/>
        <div className="row">
        {
         /*creating card components*/
         this.props.cards.map((card, index) => {
           return (<Card key={index}  {...card} />);
         })
        }
       </div>
    </div>
  );
  }
}

Remember.propTypes ={
  cards: PropTypes.array.isRequired
}

export const mapStateToProps = (state) => {
  return {
    cards: state.cards
  };
}

export default connect( mapStateToProps)(Remember);
