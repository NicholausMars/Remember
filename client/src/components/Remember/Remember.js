import React, {Component} from 'react';
import { connect } from 'react-redux';
import {deleteCard} from '../../actions';
import {Card} from './Card';
import Form from './Form';
import PropTypes from 'prop-types';
/*
Need to Give every card a unique id

*/
export class Remember extends Component{
  render(){
    // console.log(this.props);
    return (
      <div>
        <h1 className="heading-primary">Remember</h1>
        <Form />
        <hr className="u-mt-sm u-mb-sm"/>
        <div className="row">
        {
         /*creating card components*/
         this.props.cards.map((card, index) => {
           return (<Card  key={card.id}  {...card} deleteCard={this.props.deleteCard} />);
         })
        }
       </div>
    </div>
  );
  }
}

Remember.propTypes ={
  cards: PropTypes.array.isRequired
}

export const mapStateToProps = (state) => {
  return {
    cards: state.cards
  };
}

export default connect(mapStateToProps,{deleteCard})(Remember);
// export default connect(null, {deleteCard})(Card);
