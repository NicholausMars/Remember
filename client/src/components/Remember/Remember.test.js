import React from 'react';
import {shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Rememberprops as props} from '../../data/constance';
import {Remember, mapStateToProps} from './Remember';
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

describe('Remember', ()=>{
  const remember = shallow(<Remember {...props}/>);
  test('Remember render correctly', () => expect(remember).toMatchSnapshot());
  test('render header title',()=>expect(remember.find('h1').text()).toEqual('Remember'));
  test('render Form Component',()=>expect(remember.find('Connect(Form)').exists()).toBe(true));
  test('should show array of cards added', ()=> expect(mapStateToProps(props)).toEqual({cards: props.cards}));
});
