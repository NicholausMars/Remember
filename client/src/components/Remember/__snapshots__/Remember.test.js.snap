// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remember Remember render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Remember
    cards={
        Array [
            Object {
              "description": "DESCRIPTION1",
              "title": "TITLE1",
            },
            Object {
              "description": "DESCRIPTION2",
              "title": "TITLE2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="heading-primary"
>
          Remember
</h1>,
        <Connect(Form) />,
        <hr
          className="u-mt-sm u-mb-sm"
/>,
        <div
          className="row"
>
          <Unknown
                    description="DESCRIPTION1"
                    title="TITLE1"
          />
          <Unknown
                    description="DESCRIPTION2"
                    title="TITLE2"
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Remember",
          "className": "heading-primary",
        },
        "ref": null,
        "rendered": "Remember",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "u-mt-sm u-mb-sm",
        },
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              description="DESCRIPTION1"
              title="TITLE1"
/>,
            <Unknown
              description="DESCRIPTION2"
              title="TITLE2"
/>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "description": "DESCRIPTION1",
              "title": "TITLE1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "description": "DESCRIPTION2",
              "title": "TITLE2",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="heading-primary"
>
            Remember
</h1>,
          <Connect(Form) />,
          <hr
            className="u-mt-sm u-mb-sm"
/>,
          <div
            className="row"
>
            <Unknown
                        description="DESCRIPTION1"
                        title="TITLE1"
            />
            <Unknown
                        description="DESCRIPTION2"
                        title="TITLE2"
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Remember",
            "className": "heading-primary",
          },
          "ref": null,
          "rendered": "Remember",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "u-mt-sm u-mb-sm",
          },
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                description="DESCRIPTION1"
                title="TITLE1"
/>,
              <Unknown
                description="DESCRIPTION2"
                title="TITLE2"
/>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "description": "DESCRIPTION1",
                "title": "TITLE1",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "description": "DESCRIPTION2",
                "title": "TITLE2",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
